src/components/Asset/Edit/FormAddService.tsx

```diff
@@ -0,0 +1,119 @@
+import { ReactElement } from 'react'
+import { Field, Form, useFormikContext } from 'formik'
+import Input from '@shared/FormInput'
+import FormActions from './FormActions'
+import { getFieldContent } from '@utils/form'
+import consumerParametersContent from '../../../../content/publish/consumerParameters.json'
+import { ServiceEditForm } from './_types'
+import IconDownload from '@images/download.svg'
+import IconCompute from '@images/compute.svg'
+import FormEditComputeService from './FormEditComputeService'
+import { defaultServiceComputeOptions } from './_constants'
+
+export default function FormAddService({
+  data,
+  chainId
+}: {
+  data: FormFieldContent[]
+  chainId: number
+}): ReactElement {
+  const { values, setFieldValue } = useFormikContext<ServiceEditForm>()
+
+  const accessTypeOptionsTitles = getFieldContent('access', data).options
+
+  const accessTypeOptions = [
+    {
+      name: 'access-download',
+      value: 'access',
+      title: accessTypeOptionsTitles[0],
+      icon: <IconDownload />,
+      // BoxSelection component is not a Formik component
+      // so we need to handle checked state manually.
+      checked: values.access === 'access'
+    },
+    {
+      name: 'access-compute',
+      value: 'compute',
+      title: accessTypeOptionsTitles[1],
+      icon: <IconCompute />,
+      checked: values.access === 'compute'
+    }
+  ]
+
+  return (
+    <Form style={{ margin: 20 }}>
+      <Field {...getFieldContent('name', data)} component={Input} name="name" />
```

Comment:
you can do a funtion renderField(fieldname)
and in Form do somenthing like:
<Form style={{ margin: 20 }}>
{renderFiled(name)}
{renderField(description}
...

Reply:
I would rather keep it like this because the same way it is done in publish flow

Reply:
ok